[BUG] Cannot find module 'playwright/jsx-runtime' when running tests through vscode extension

**Context:**
- Playwright Version: 1.38.
- Operating System: MacOS
- Node.js version:  18
- Visual Studio Code version: 1.82.2
- Playwright for VSCode extension version: v1.0.15
- Browser: All
- Extra:  I am importing some JSX components

**Code Snippet**

```
Error: Cannot find module 'playwright/jsx-runtime'
Require stack:
- /Users/rsingh/Projects/subskribe/billy-ui-ng/src/components/snackBar/billySnackBar.tsx
- /Users/rsingh/Projects/subskribe/billy-ui-ng/src/components/SnackbarHandler/SnackbarAlertContainer.tsx
- /Users/rsingh/Projects/subskribe/billy-ui-ng/src/components/SnackbarHandler/SnackbarHandler.tsx
- /Users/rsingh/Projects/subskribe/billy-ui-ng/src/components/error/BillyErrorSnackbarDisplay.tsx
- /Users/rsingh/Projects/subskribe/billy-ui-ng/src/components/ErrorHandler/ErrorHandler.tsx
- /Users/rsingh/Projects/subskribe/billy-ui-ng/src/components/data/billyUrql.ts
- /Users/rsingh/Projects/subskribe/billy-ui-ng/src/generated/graphql.tsx
- /Users/rsingh/Projects/subskribe/billy-ui-ng/playwright/pageObjectModels/playwright-authApi.ts
- /Users/rsingh/Projects/subskribe/billy-ui-ng/playwright/fixtures/base.fixture.ts
- /Users/rsingh/Projects/subskribe/billy-ui-ng/playwright/local/customFields.spec.ts
- /Users/rsingh/Projects/subskribe/billy-ui-ng/node_modules/.pnpm/[playwright@1.38.1](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)/node_modules/playwright/lib/transform/transform.js
- /Users/rsingh/Projects/subskribe/billy-ui-ng/node_modules/.pnpm/[playwright@1.38.1](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)/node_modules/playwright/lib/common/config.js
- /Users/rsingh/Projects/subskribe/billy-ui-ng/node_modules/.pnpm/[playwright@1.38.1](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)/node_modules/playwright/lib/reporters/json.js
- /Users/rsingh/Projects/subskribe/billy-ui-ng/node_modules/.pnpm/[playwright@1.38.1](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)/node_modules/playwright/lib/reporters/html.js
- /Users/rsingh/Projects/subskribe/billy-ui-ng/node_modules/.pnpm/[playwright@1.38.1](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)/node_modules/playwright/lib/runner/reporters.js
- /Users/rsingh/Projects/subskribe/billy-ui-ng/node_modules/.pnpm/[playwright@1.38.1](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)/node_modules/playwright/lib/runner/runner.js
- /Users/rsingh/Projects/subskribe/billy-ui-ng/node_modules/.pnpm/[playwright@1.38.1](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)/node_modules/playwright/lib/cli.js
- /Users/rsingh/Projects/subskribe/billy-ui-ng/node_modules/.pnpm/[playwright@1.38.1](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)/node_modules/playwright/cli.js
- /Users/rsingh/Projects/subskribe/billy-ui-ng/node_modules/.pnpm/@[playwright+test@1.38.1](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html)/node_modules/@playwright/test/cli.js
```

I get a `Error: Cannot find module 'playwright/jsx-runtime'` when importing a `tsx` component  however it's quite far down the stack. However it doesn't show up when I run playwright from the CLI

Hello,

I've managed to create a minimal reproduction of this issue inside this repository:

https://github.com/woahitsraj/Playwright-failing-vscode-example

Steps:

1. Run `pnpm i`
2. Run `npx playwright test`
3. Observe tests run fine with no issue
4. Open vscode
5. Open "Testing" tab in Playwright for Vscode Extension
6. Try to run tests
7. Receive error
```
Error: Cannot find module 'playwright/jsx-runtime'
```

--- a/packages/playwright/bundles/babel/src/babelBundleImpl.ts
+++ b/packages/playwright/bundles/babel/src/babelBundleImpl.ts
@@ -65,7 +65,7 @@ function babelTransformOptions(isTypeScript: boolean, isModule: boolean, plugins
   // Support JSX/TSX at all times, regardless of the file extension.
   plugins.push([require('@babel/plugin-transform-react-jsx'), {
     runtime: 'automatic',
-    importSource: 'playwright'
+    importSource: require('path').dirname(require.resolve('playwright')),
   }]);
 
   if (!isModule) {

--- a/packages/playwright/bundles/babel/src/babelBundleImpl.ts
+++ b/packages/playwright/bundles/babel/src/babelBundleImpl.ts
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 
+import path from 'path';
 import type { BabelFileResult, NodePath, PluginObj, TransformOptions } from '@babel/core';
 import type { TSExportAssignment, ImportDeclaration } from '@babel/types';
 import type { TemplateBuilder } from '@babel/template';
@@ -65,7 +66,7 @@ function babelTransformOptions(isTypeScript: boolean, isModule: boolean, plugins
   // Support JSX/TSX at all times, regardless of the file extension.
   plugins.push([require('@babel/plugin-transform-react-jsx'), {
     runtime: 'automatic',
-    importSource: require('path').dirname(require.resolve('playwright')),
+    importSource: path.dirname(require.resolve('playwright')),
   }]);
 
   if (!isModule) {

--- a/tests/installation/fixture-scripts/read-json-report.js
+++ b/tests/installation/fixture-scripts/read-json-report.js
@@ -3,11 +3,20 @@ if (report.suites[0].specs[0].title !== 'sample test') {
   console.log(`Wrong spec title`);
   process.exit(1);
 }
+
 const projects = report.suites[0].specs[0].tests.map(t => t.projectName).sort();
-if (projects.length !== 3 || projects[0] !== 'chromium' || projects[1] !== 'firefox' || projects[2] !== 'webkit') {
-  console.log(`Wrong browsers`);
-  process.exit(1);
+if (process.argv.slice(3).includes('--validate-chromium-project-only')) {
+  if (projects.length !== 1 || projects[0] !== 'chromium') {
+    console.log(`Wrong browsers`);
+    process.exit(1);
+  }
+} else {
+  if (projects.length !== 3 || projects[0] !== 'chromium' || projects[1] !== 'firefox' || projects[2] !== 'webkit') {
+    console.log(`Wrong browsers`);
+    process.exit(1);
+  }
 }
+
 for (const test of report.suites[0].specs[0].tests) {
   if (test.results[0].status !== 'passed') {
     console.log(`Test did not pass`);

--- a/tests/installation/npmTest.ts
+++ b/tests/installation/npmTest.ts
@@ -119,8 +119,10 @@ export const test = _test
       }],
       writeFiles: async ({ tmpWorkspace }, use) => {
         await use(async (nameToContents: Record<string, string>) => {
-          for (const [name, contents] of Object.entries(nameToContents))
+          for (const [name, contents] of Object.entries(nameToContents)) {
+            await fs.promises.mkdir(path.join(tmpWorkspace, path.dirname(name)), { recursive: true });
             await fs.promises.writeFile(path.join(tmpWorkspace, name), contents);
+          }
         });
       },
       tmpWorkspace: async ({}, use) => {

--- a/tests/installation/playwright-component-testing.spec.ts
+++ b/tests/installation/playwright-component-testing.spec.ts
@@ -68,6 +68,6 @@ test('pnpm: JSX inside a @playwright/test should work', async ({ exec, tmpWorksp
       });
     `,
   });
-  await exec(`node node_modules/@playwright/test/cli.js test --browser=all --reporter=list,json example.spec.ts`, { env: { PLAYWRIGHT_JSON_OUTPUT_NAME: 'report.json' } });
-  await exec('node read-json-report.js', path.join(tmpWorkspace, 'report.json'));
+  await exec(`node node_modules/@playwright/test/cli.js test --browser=chromium --reporter=list,json example.spec.ts`, { env: { PLAYWRIGHT_JSON_OUTPUT_NAME: 'report.json' } });
+  await exec('node read-json-report.js', path.join(tmpWorkspace, 'report.json'), '--validate-chromium-project-only');
 });

--- a/tests/installation/playwright-component-testing.spec.ts
+++ b/tests/installation/playwright-component-testing.spec.ts
@@ -46,8 +46,8 @@ test('pnpm: @playwright/experimental-ct-react should work', async ({ exec, tmpWo
       });
     `,
   });
-  await exec('pnpm exec playwright test -c . --browser=all --reporter=list,json example.spec.tsx', { env: { PLAYWRIGHT_JSON_OUTPUT_NAME: 'report.json' } });
-  await exec('node read-json-report.js', path.join(tmpWorkspace, 'report.json'));
+  await exec('pnpm exec playwright test -c . --browser=chromium --reporter=list,json example.spec.tsx', { env: { PLAYWRIGHT_JSON_OUTPUT_NAME: 'report.json' } });
+  await exec('node read-json-report.js', path.join(tmpWorkspace, 'report.json'), '--validate-chromium-project-only');
 });
 
 test('pnpm: JSX inside a @playwright/test should work', async ({ exec, tmpWorkspace, writeFiles }) => {

--- a/tests/installation/playwright-component-testing.spec.ts
+++ b/tests/installation/playwright-component-testing.spec.ts
@@ -1,0 +1,73 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { test } from './npmTest';
+import path from 'path';
+
+test('pnpm: @playwright/experimental-ct-react should work', async ({ exec, tmpWorkspace, writeFiles }) => {
+  await exec('pnpm add @playwright/experimental-ct-react react react-dom');
+  await exec('pnpm exec playwright install');
+  await writeFiles({
+    'playwright.config.ts': `
+      import { defineConfig } from '@playwright/experimental-ct-react';
+      export default defineConfig({});
+    `,
+    'playwright/index.html': `<script type="module" src="./index.js"></script>`,
+    'playwright/index.js': ``,
+    'Button.tsx': `
+      export function Button({ onClick }) {
+        return <button onClick={onClick}>Submit</button>;
+      }
+    `,
+    'example.spec.tsx': `
+      import { test, expect } from '@playwright/experimental-ct-react';
+      import { Button } from './Button';
+
+      test('sample test', async ({ page, mount }) => {
+        let clicked = false;
+        const component = await mount(
+          <Button title='Submit' onClick={() => { clicked = true }}></Button>
+        );
+        await expect(component).toContainText('Submit');
+        await component.click();
+        expect(clicked).toBeTruthy();
+      });
+    `,
+  });
+  await exec('pnpm exec playwright test -c . --browser=all --reporter=list,json example.spec.tsx', { env: { PLAYWRIGHT_JSON_OUTPUT_NAME: 'report.json' } });
+  await exec('node read-json-report.js', path.join(tmpWorkspace, 'report.json'));
+});
+
+test('pnpm: JSX inside a @playwright/test should work', async ({ exec, tmpWorkspace, writeFiles }) => {
+  await exec('pnpm add @playwright/test');
+  await exec('pnpm exec playwright install');
+  await writeFiles({
+    'Button.tsx': `
+      export function Button({ onClick }) {
+        return <button onClick={onClick}>Submit</button>;
+      }
+    `,
+    'example.spec.ts': `
+      import { test, expect } from '@playwright/test';
+      import { Button } from './Button';
+
+      test('sample test', async ({ page }) => {
+        expect(Button).toEqual(expect.any(Function));
+      });
+    `,
+  });
+  await exec(`node node_modules/@playwright/test/cli.js test --browser=all --reporter=list,json example.spec.ts`, { env: { PLAYWRIGHT_JSON_OUTPUT_NAME: 'report.json' } });
+  await exec('node read-json-report.js', path.join(tmpWorkspace, 'report.json'));
+});

